### PR TITLE
Upgrade to appwrite 1.0

### LINKED ISSUE #4269: 🪄 Upgrade ’demo-almost-netflix-for-flutter’ to Appwrite 1.0
# 💭 Introduction

Appwrite is all about open source and community 👥 To emphasize this, we want to create **the best** developer experience possible by providing enough guides and examples for new Appwriters.

Your task is to upgrade our demo application **appwrite/demo-almost-netflix-for-flutter** to the latest Appwrite version, which is 1.0. All of the functionality of the demo application must stay the same. 

### ✅ Task Summary

- [ ]  Ask to be assigned to the issue.
- [ ]  Wait to be assigned.
- [ ]  Upgrade appwrite/demo-almost-netflix-for-flutter to support Appwrite 1.0.
- [ ]  Submit a pull request in **appwrite/demo-almost-netflix-for-flutter**.

If you have questions, need any help, or just want to hang out, make sure to join us on our [Discord server](https://appwrite.io/discord).

**Happy Appwriting!**

### PR DESCRIPTION
**Pull request by letsintegreat**
## What does this PR do?

Upgrades the app to use appwrite 1.0

## Test Plan

I have used this [blogpost](https://dev.to/appwrite/did-we-just-build-a-netflix-clone-with-appwrite-28ok) to setup the project, and tested the app in an emulator.

Here are some ids used:
projectID: almostNetflix2
databaseID: default2
collectionID: movies, watchlists
bucketID: default1

## Related PRs and Issues

Closes https://github.com/appwrite/appwrite/issues/4269

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
@letsintegreat Great PR! 🤯 Would you please also address the following:

- [ ] Make use of the `ID`, `Permission`, and `Role` helpers where applicable rather than using magic strings
- [ ] Remove `createdAt` from the watchlist collection since the attribute is built in now.
- [ ] genres, tags, and cast in Entry should be lists

The app doesn't run smoothly for me and crashes quite often. Does that happen for you?

**Comment by PineappleIOnic** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Comment by letsintegreat** [comment]
@stnguyen90 Hey, sorry for the late reply, I got busy with college stuff. Anyways, now I am back and will try to make the PR merge-ready ASAP.

So, I am working to address the last 2 points. Though I am not sure what ID, Permission, and Role helpers are. Can you please elaborate on the first point while I work on the others? Thanks.

**Comment by stnguyen90** [comment]
> So, I am working to address the last 2 points. Though I am not sure what ID, Permission, and Role helpers are. Can you please elaborate on the first point while I work on the others? Thanks.

For example, see https://github.com/appwrite/playground-for-flutter/blob/ceea7c3b491506d4c1f7c8da5149153824092eec/lib/main.dart#L103-L108.

**Comment by letsintegreat** [comment]
> @letsintegreat Great PR! 🤯 Would you please also address the following:
> 
> * [ ]  Make use of the `ID`, `Permission`, and `Role` helpers where applicable rather than using magic strings
> * [ ]  Remove `createdAt` from the watchlist collection since the attribute is built in now.
> * [ ]  genres, tags, and cast in Entry should be lists
> 
> The app doesn't run smoothly for me and crashes quite often. Does that happen for you?

@stnguyen90 Hey, I have addressed all the 3 points, please review the changes!

File: lib/data/entry.dart [hunk 1]
```diff
@@ -17,13 +17,13 @@ class Entry {
   final String ageRestriction;
   final Duration durationMinutes;
   final String thumbnailImageId;
-  final String genres;
-  final String tags;
+  final List<String> genres;
```
Comment (stnguyen90) [inline lib/data/entry.dart:20]: After logging in I got this error:

> `_TypeError (type 'List<dynamic>' is not a subtype of type 'List<String>')`

File: lib/providers/watchlist.dart [hunk 2]
```diff
@@ -56,12 +59,12 @@ class WatchListProvider extends ChangeNotifier {
     final user = await this.user;
 
     var result = await ApiClient.database.createDocument(
+        databaseId: _databaseId,
         collectionId: _collectionId,
         documentId: 'unique()',
```
Comment (stnguyen90) [inline lib/providers/watchlist.dart]: Please replace this with `ID.unique()`

File: lib/providers/account.dart [hunk 3]
```diff
@@ -16,8 +16,8 @@ import 'package:flutter/material.dart';
 import 'package:netflix_clone/data/store.dart';
 
 class AccountProvider extends ChangeNotifier {
-  User? _current;
-  User? get current => _current;
+  Account? _current;
+  Account? get current => _current;
```
Comment (stnguyen90) [inline lib/providers/account.dart:21]: Please replace the `unique()` in this file too.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: lib/data/entry.dart (hunk 1 above)
Comment (letsintegreat) [reply lib/data/entry.dart:20]: @stnguyen90 I am not able to reproduce this error? I have addressed the rest of the two requests though.

Comment (stnguyen90) [reply lib/data/entry.dart:20]: @letsintegreat is `genres` a String Array attribute in your instance? Do you have documents with non empty `genres`?

Comment (letsintegreat) [reply lib/data/entry.dart:20]: @stnguyen90 Yes and Yes...

image

image

Comment (stnguyen90) [reply lib/data/entry.dart:20]: @letsintegreat, what version of Flutter do you have? I have:

```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.9, on macOS 12.6.2 21G320 darwin-arm, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.74.3)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!
```

`data` is this for me:

image

I think this type of error is pretty common. See this sample in [Zapp](https://zapp.run/edit/dart-z11y062t11z0?lazy=false&split=50&entry=lib/main.dart&file=lib/main.dart).

```dart
import 'dart:convert';

void main() {
  String body = '{"genres": ["Action", "Adventure", "Fantasy"]}';
  final data = jsonDecode(body);
  
  print(data);
  print(data['genres']);
  print(data['genres'].runtimeType);
  
  List<String> genres = data['genres'];
}
```

Comment (letsintegreat) [reply lib/data/entry.dart:20]: @stnguyen90 Ah, I am sorry I was testing the app on a different branch. Yes, you are right, this was indeed a problem. I have fixed this and a few other bugs, and have tested the app thoroughly this time *in the correct branch*. Sorry for the inconvenience!

**Review by stnguyen90 — APPROVED** [review]
Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**